presto-function-namespace-managers/src/main/java/com/facebook/presto/functionNamespace/json/JsonBasedUdfFunctionMetaInfo.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.functionNamespace.json;
+
+import com.facebook.presto.spi.function.RoutineCharacteristics;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.IntStream;
+
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static java.util.Objects.requireNonNull;
+
+public class JsonBasedUdfFunctionMetaInfo
+{
+    private final String docString;
+    private final String outputType;
+    private final Optional<List<String>> paramNames;
+    private final List<String> paramTypes;
+    private final String schema;
+    private final RoutineCharacteristics routineCharacteristics;
+
+    @JsonCreator
+    public JsonBasedUdfFunctionMetaInfo(
```

Comment:
I feel like Metadata is a more common term than MetaInfo

Reply:
Done